plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/data/ExecuteBatchImpl.java

```diff
@@ -193,7 +193,12 @@ DBCStatistics processBatch(@NotNull DBCSession session, @Nullable List<DBEPersis
                             }
                         }
                     } else {
-                        String queryString = formatQueryParameters(session, statement.getQueryString(), handlers, rowValues);
+                        String queryString;
+                        if (statement instanceof DBCParameterizedStatement parameterizedStatement) {
+                            queryString = parameterizedStatement.getFormattedQuery();
```

Comment:
parameterizedStatement.getFormattedQuery() uses a silly algorithm. It is not always correct.
Now we switch to it always. Are you sure it is a good solution? Probably DBCParameterizedStatement  should be used for virtual keys only